clients/src/main/java/org/apache/kafka/clients/consumer/internals/AbstractCoordinator.java

```diff
@@ -323,11 +323,17 @@ private synchronized void disableHeartbeatThread() {
     }
 
     private void closeHeartbeatThread() {
```

Comment:
I had had the same thought but wasn't sure exactly what set of findbugs issues this code was triggering. I agree that's a cleaner, simpler solution if it works.